lib/cdo/rack/attack.rb

```diff
@@ -0,0 +1,59 @@
+require 'rack/attack'
+require 'active_support/core_ext/numeric/time'
+
+class Rack::Attack
+  # maximum requests per minute over a 4-minute stretch.
+  MAX_TABLE_READS_PER_MIN = 2 * 60
+  MAX_TABLE_WRITES_PER_MIN = 4 * 60
+  MAX_PROPERTY_READS_PER_MIN = 4 * 60
+  MAX_PROPERTY_WRITES_PER_MIN = 4 * 60
+
+  redis_url = CDO.geocoder_redis_url || 'redis://localhost:6379'
+  cache.store = StoreProxy::RedisStoreProxy.new(Redis.new(url: redis_url))
+
+  def self.limits(max_per_min)
+    # allow 4x the max rate over 15s, 2x the max rate over 60s, and 1x the max rate over 4min.
+    # this gives the experience of exponential backoff when limits are exceeded.
```

Comment:
Currently RackAttack returns status code 429 'Retry later'. This is configurable. App Lab prints to the console: `error writing value: unexpected http status 429`